### DISCUSSION TITLE
location of temporary runtime files

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#22101] [Q&A General] [answered]
Hi everyone,

When running MOOSE, it creates temporary files along the lines of "tmp_jit_HOyNo6.so".  Is it possible to specify the location of those files?

a

**Comment by GiudGiud** [comment, marked as answer]
Hello

I dont know that you can. @dschwen will know for sure
There's some paramters to get rid of them, disable_fpoptimizer I think
they are under Advanced
https://mooseframework.inl.gov/source/materials/DerivativeParsedMaterial.html

Guillaume

**Reply by WilkAndy** [reply]
Thank you @GiudGiud

**Reply by milljm** [reply]
May I ask why you need this? If you have the ability to control where these files are created, it may help others.
Others that have in the past, wished to maintain a centralized location of MOOSE source where multiple people 'build' from, yet don't have write access.

**Reply by WilkAndy** [reply]
My HPC staff asked whether this was possible, because they think that writing lots of these temporary files causes the filesystem to slowdown for everyone (I'm running many MOOSE jobs).  If i could write to another filesystem then it would help address that slowdown.

a
